### PR TITLE
support ES6+ syntax by babel

### PR DESCRIPTION
**Pull request by fundon**

### CODE REVIEW

**Comment by fundon** [comment]
@sdepold I submited a new and clean PR.

File: lib/assets/migrations/create-table.es6.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          allowNull: false,
+          type: Sequelize.DATE
+        },
+
+        updatedAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        }
+      })
+      .done(done);
```
Comment (sdepold) [inline lib/assets/migrations/create-table.es6.js]: This line has to be removed

File: lib/assets/models/index.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ var path      = require('path');
 var Sequelize = require('sequelize');
 var basename  = path.basename(module.filename);
 var env       = process.env.NODE_ENV || 'development';
-var config    = require(<%= configFile %>)[env];
```
Comment (sdepold) [inline lib/assets/models/index.js]: This change will break https://github.com/sequelize/cli/blob/22f208e7fda83268a8ae003b1074a12573befd83/lib/helpers/init-helper.js#L72

Comment (sdepold) [inline lib/assets/models/index.js]: And the fact that the build is still green makes me think, that we should add a test for that

**Comment by sdepold** [comment]
Is it a common thing to add `es6` to the file extension? I think it's odd :)

File: test/support/assets/migrations/20111117063700-createPerson.es6.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+export default {
+
```
Comment (sdepold) [inline test/support/assets/migrations/20111117063700-createPerson.es6.js]: Is this line break personal taste or a common thing in ES6 land? I'd prefer the removal of the padding lines in a block.

**Comment by sdepold** [comment]
Good job :) I will take another final look later :)

**Pull request closed without merging**